genequery-rest/src/main/kotlin/gq/rest/validators/Validators.kt

```diff
@@ -61,3 +62,25 @@ class EnrichmentRequestFormValidator : Validator {
 
     override fun supports(clazz: Class<*>?) = GeneSetEnrichmentController.EnrichmentRequestForm::class.java.equals(clazz)
 }
+
+
+class OverlapRequestFormValidator : Validator {
+    override fun validate(target: Any?, errors: Errors?) {
+        val overlapRequestForm = target as OverlapController.OverlapRequestForm
+        val speciesValidator = SpeciesValidator()
+
+        if (validateIsNotNullAndNotBlank(overlapRequestForm.speciesFrom, errors, "speciesFrom")) {
+            ValidationUtils.invokeValidator(speciesValidator, overlapRequestForm.speciesFrom, errors)
+        }
+        if (validateIsNotNullAndNotBlank(overlapRequestForm.speciesTo, errors, "speciesTo")) {
+            ValidationUtils.invokeValidator(speciesValidator, overlapRequestForm.speciesTo, errors)
+        }
+
+        validateIsNotNullAndNotBlank(overlapRequestForm.genes, errors, "genes")
+
+        //TODO: probably should validate it the same way we validate speciesFrom and speciesTo
```

Comment:
We check if speciesFrom/speciesTo in request is actually species (hs/mm/rt). Shouldn't we do some more complex validation of moduleName as well? Like check if it fills [these requirements](https://github.com/ctlab/genequery-kotlin/blob/master/genequery-core/src/main/kotlin/gq/core/data/GQModule.kt#L61-L63).